app/src/main/java/com/zulip/android/activities/ZulipActivity.java

```diff
@@ -485,6 +485,10 @@ public boolean setViewValue(View view, Cursor cursor, int columnIndex) {
                         }
                         return true;
                     case R.id.nameTV:
+                        //currently emojis are in png format, add others if new format(jpg, jpeg etc) is introduced
+                        if (!personTable && name.endsWith(".png")) {
```

Comment:
Another way for doing this is to remove ".png" when setting up Emoji in [ZulipApp.java](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/ZulipApp.java) at setupEmoji()

And then while setting Image Drawable for Emoji by adding ".png" after name in [ZulipActivity.java](https://github.com/zulip/zulip-android/blob/master/app/src/main/java/com/zulip/android/activities/ZulipActivity.java) at combinedAdapter.setViewBinder()

**ZulipApp.java**
`for (String newEmoji : emojis) {
                                newEmoji = newEmoji.replace(".png" , "");
                                dao.create(new Emoji(newEmoji));
                            }`

**ZulipActivity.java**

`Drawable drawable = Drawable.createFromStream(getApplicationContext().getAssets().open("emoji/" + name +".png"),
                                        "emoji/" + name + ".png");
                                ((ImageView) view).setImageDrawable(drawable);`


But either way we are manupulating string ... :smile_cat: